components/camel-spring-xml/pom.xml

```diff
@@ -137,6 +137,15 @@
             <artifactId>derby</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+        </dependency>
+
+        <dependency>
```

Comment:
I wonder what makes saaj-impl needed?

Reply:
I don't remember right now what's requiring it (I think it's a test case).

saaj-impl is also causing problems for Java 17. This version is very outdated.

Reply:
Can you try testing, eg set it to test scope / also try without so we can find out which test, and then add a comment in these testst that they use saaj so we know in the future

Reply:
Absolutely, I will take a look at this.

Reply:
@davsclaus great catch! So, `saaj-impl` is not needed on the `camel-spring-xml`. It' actually needed by `camel-spring-ws` tests. Marking it as `test` scope seems to be enough.